frontend/www/js/omegaup/components/contest/Intro.vue

```diff
@@ -0,0 +1,311 @@
+                ></omegaup-countdown>
+              </span>
+            </p>
+          </div>
+
+          <div
+            v-if="
+              now > contest.finish_time.getTime() ||
+                now > contest.start_time.getTime()
+            "
+          >
+            <form
+              v-on:submit.prevent="onStartContest"
+              v-if="
+                contest.admission_mode !== 'registration' ||
+                  contest.user_registration_accepted
+              "
+            >
+              <p
+                v-if="
+                  !needsBasicInformation && requestsUserInformation === 'no'
+                "
+              >
+                {{ T.aboutToStart }}
+              </p>
+              <p
+                v-if="needsBasicInformation"
+                v-html="T.contestBasicInformationNeeded"
+              ></p>
+              <template v-if="requestsUserInformation !== 'no'">
+                <p v-html="consentHtml"></p>
+                <p>
+                  <label>
+                    <input
+                      type="radio"
+                      v-model="shareUserInformation"
+                      v-bind:value="true"
+                    />
+                    {{ T.wordsYes }}
+                  </label>
+                  <label>
+                    <input
+                      type="radio"
+                      v-model="shareUserInformation"
+                      v-bind:value="false"
+                    />
+                    {{ T.wordsNo }}
+                  </label>
+                </p>
+              </template>
+              <button
+                type="submit"
+                v-bind:disabled="isButtonDisabled"
+                class="btn btn-primary btn-lg start-contest-submit"
+              >
+                {{ T.startContest }}
+              </button>
+            </form>
+
+            <!-- Must register -->
+            <form v-else="" v-on:submit.prevent="onRequestAccess">
+              <template v-if="!contest.user_registration_requested">
+                <p>{{ T.mustRegisterToJoinContest }}</p>
+                <button type="submit" class="btn btn-primary btn-lg">
+                  {{ T.registerForContest }}
+                </button>
+              </template>
+              <!-- Registration pending -->
+              <div v-else-if="!contest.user_registration_answered">
+                <p>{{ T.registrationPending }}</p>
+              </div>
+              <!-- Registration denied -->
+              <div v-else-if="!contest.user_registration_answered === false">
+                <p>{{ T.registrationDenied }}</p>
+              </div>
+            </form>
+          </div>
+        </template>
+
+        <template v-else="">
+          <!-- Must login to do anything -->
+          <div class="panel">
+            <p>{{ T.mustLoginToJoinContest }}</p>
+            <a
+              v-bind:href="`/login/?redirect=${requestURI}`"
```

Comment:
```suggestion
              v-bind:href="`/login/?redirect=${encodeURIComponent(window.location.href)}`"
```

esto no funciona? como que quiero ver si se puede evitar mandar `requestURI` desde el servidor porque esa información se puede obtener localmente.

Reply:
Si funciona, lo tuve que mover a una propiedad computada porque de esa forma se quejaba `TypeScript`.